src/ncepbufr/prepbufr_adpsfc_api.py

```diff
@@ -0,0 +1,234 @@
+    virtualtemperatureqm.atts.create('units', ioda.Types.str).writeVector.str(['1'])
+    virtualtemperatureqm.atts.create('long_name', ioda.Types.str).writeVector.str(['Virtual Temperature Quality Marker'])
+
+    stationpressureqm = g.vars.create('QualityMarker/stationPressure', ioda.Types.int, scales=[dim_location], params=pint)
+    stationpressureqm.atts.create('units', ioda.Types.str).writeVector.str(['1'])
+    stationpressureqm.atts.create('long_name', ioda.Types.str).writeVector.str(['Station Pressure Quality Marker'])
+
+    specifichumidityqm = g.vars.create('QualityMarker/specificHumidity', ioda.Types.int, scales=[dim_location], params=pint)
+    specifichumidityqm.atts.create('units', ioda.Types.str).writeVector.str(['1'])
+    specifichumidityqm.atts.create('long_name', ioda.Types.str).writeVector.str(['Specific Humidity Quality Marker'])
+
+    windnorthwardqm = g.vars.create('QualityMarker/windNorthward', ioda.Types.int, scales=[dim_location], params=pint)
+    windnorthwardqm.atts.create('units', ioda.Types.str).writeVector.str(['1'])
+    windnorthwardqm.atts.create('long_name', ioda.Types.str).writeVector.str(['Northward Quality Marker'])
+
+    windeastwardqm = g.vars.create('QualityMarker/windEastward', ioda.Types.int, scales=[dim_location], params=pint)
+    windeastwardqm.atts.create('units', ioda.Types.str).writeVector.str(['1'])
+    windeastwardqm.atts.create('long_name', ioda.Types.str).writeVector.str(['Eastward Wind Quality Marker'])
+
+    # Write the data to the variables
+    print("Write data to variables")
+    longitude.writeNPArray.float(lon.flatten())
+    latitude.writeNPArray.float(lat.flatten())
+    datetime.writeNPArray.int64(dhr.flatten())
+    stationelevation.writeNPArray.float(elv.flatten())
+    temperatureeventcode.writeNPArray.int(tpc.flatten())
+
+    airtemperature.writeNPArray.float(tob.flatten())
+    virtualtemperature.writeNPArray.float(tvo.flatten())
+    stationpressure.writeNPArray.float(pob.flatten())
+    pressure.writeNPArray.float(pob.flatten())
+    windnorthward.writeNPArray.float(vob.flatten())
+    windeastward.writeNPArray.float(uob.flatten())
+    specifichumidity.writeNPArray.float(qob.flatten())
+
+    airtemperatureqm.writeNPArray.int(tobqm.flatten())
+    virtualtemperatureqm.writeNPArray.int(tvoqm.flatten())
+    stationpressureqm.writeNPArray.int(pobqm.flatten())
+    specifichumidityqm.writeNPArray.int(qobqm.flatten())
+    windeastwardqm.writeNPArray.int(uobqm.flatten())
+    windnorthwardqm.writeNPArray.int(vobqm.flatten())
+
+    print("end")
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    description = (
+        'Reads NCEP PREPBUFR formated ADPsurface input files'
+        '  created by split_by_subset from a PREPBUFR file'
+        '  convert into IODA formatted output files.'
+    )
+
+    optional = parser.add_argument_group(title='optional arguments')
```

Comment:
ditto optional => required
```suggestion
    required = parser.add_argument_group(title='required arguments')
```